### PR TITLE
THREESCALE-11358 update the apicast config and curl commands for echo-api

### PR DESCRIPTION
**Pull request by austincunningham**
## Issue
THREESCALE-11358
## What
update the apicast config and the curl requests in the policy dev tutorial

### CODE REVIEW

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md [hunk 1]
```diff
@@ -92,6 +99,7 @@ $ curl localhost:8080
 </body>
 </html>
 ```
+>**NOTE**: alernatively you can get the docker container ip address and curl from your local machine `APICAST_IP=$(docker inspect apicast_build_0-development-1 | yq e -P '.[0].NetworkSettings.Networks.apicast_build_0_default.IPAddress' -)` and `curl -i -H "Host: one" "http://${APICAST_IP}:8080/?user_key=0123456789"`
```
Comment (dfennessy) [inline doc/policy_tutorial/POLICY_RUN_LOCALLY.md:102]:
```suggestion
> **NOTE:**
Alternatively, you can get the docker container IP address and curl from your local machine:
> - `APICAST_IP=$(docker inspect apicast_build_0-development-1 | yq e -P '.[0].NetworkSettings.Networks.apicast_build_0_default.IPAddress' -)` 
> - `curl -i -H "Host: one" "http://${APICAST_IP}:8080/?user_key=0123456789"`
```

Comment (dfennessy) [reply doc/policy_tutorial/POLICY_RUN_LOCALLY.md:102]: Fixed the capitalization and spelling of `alernatively` and attempting to make the note into a block with the commands more clear in bullets.

**Review by dfennessy — CHANGES_REQUESTED** [review]
See comment in my suggestion :)

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md (hunk 1 above)
Comment (austincunningham) [reply doc/policy_tutorial/POLICY_RUN_LOCALLY.md]: suggestion accepted

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md [hunk 2]
```diff
@@ -124,7 +135,7 @@ if secret_header ~= self.secret then
 The policy is now executing. You must provide the secret header in the request to pass the validation. Issue the following HTTP request:
 
 ```shell
-$ curl localhost:8080 -H 'secret: mysecret'
+$ curl "http://localhost:8080/?user_key=" -H 'secret: mysecret' -H "Host: one"
 GET / HTTP/1.1
 X-Real-IP: 127.0.0.1
 Host: echo
```
Comment (tkan145) [inline doc/policy_tutorial/POLICY_RUN_LOCALLY.md:141]: This is the response from echo server? If so then the `Host` header should be `echo-api.3scale.net` right?

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md [hunk 3]
```diff
@@ -136,14 +147,14 @@ secret: mysecret
...
-GET /?myparam=myvalue HTTP/1.1
+$ curl localhost:8080?user_key=myvalue -H 'secret: mysecret'
+GET /?userkey=myvalue HTTP/1.1
 X-Real-IP: 127.0.0.1
 Host: echo
 User-Agent: curl/7.29.0
 Accept: */*
 secret: mysecret
-myparam: myvalue
+user_key: myvalue
```
Comment (tkan145) [inline doc/policy_tutorial/POLICY_RUN_LOCALLY.md:157]: Same as above

**Review by tkan145 — CHANGES_REQUESTED** [review]

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md (hunk 2 above)
Comment (austincunningham) [reply doc/policy_tutorial/POLICY_RUN_LOCALLY.md:141]: hmm actual response
```bash
curl -i -H "secret: mysecret" -H "Host: one" "http://${APICAST_IP}:8080/?user_key="

HTTP/1.1 200 OK
Server: openresty
Date: Tue, 24 Sep 2024 11:13:29 GMT
Content-Type: text/plain
Transfer-Encoding: chunked
Connection: keep-alive

GET /?user_key= HTTP/1.1
X-Real-IP: 172.22.0.1
Host: echo:8081
User-Agent: curl/8.2.1
Accept: */*
secret: mysecret
```
so not `echo-api.3scale.net` though not sure why . I will update accordingly

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md [hunk 4]
```diff
@@ -83,7 +89,8 @@ Now you have another interactive bash shell in the APIcast development container
 In the container issue the following HTTP request:
 
 ```shell
-$ curl localhost:8080
+# user_key is a required paramerter for the echoapi backend
+$ curl -H "Host: one" "http://localhost:8080/?user_key="
```
Comment (eguzki) [inline doc/policy_tutorial/POLICY_RUN_LOCALLY.md:93]: nit: I would use `--resolve` curl option

```
curl --resolve one:8080:127.0.0.1 -v "http://one:8080/?user_key"
```

Up to you, not really requesting a change here.

**Review by eguzki — APPROVED** [review]

File: doc/policy_tutorial/POLICY_RUN_LOCALLY.md (hunk 4 above)
Comment (austincunningham) [reply doc/policy_tutorial/POLICY_RUN_LOCALLY.md:93]: Interesting not an option I have tried before with curl. 
A kind of dns resolver for curl .
Does it make it more or less understandable :thinking: 
Think I will leave as is

**Pull request merged**